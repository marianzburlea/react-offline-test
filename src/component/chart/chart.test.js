import React from 'react'
import { create } from 'react-test-renderer'
import Chart from '.'

describe('@Chart', () => {
  it('should match the snapshot', () => {
    const data = [
      {fuel: 'solar', perc: 75},
      {fuel: 'wind', perc: 25},
    ]
    const chart = create(<Chart data={data} />)
    expect(chart.toJSON()).toMatchSnapshot()
  })
})
