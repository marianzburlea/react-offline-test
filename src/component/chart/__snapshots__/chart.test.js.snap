// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@Chart should match the snapshot 1`] = `
<div
  className="recharts-wrapper"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseMove={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "cursor": "default",
      "height": 300,
      "position": "relative",
      "width": 500,
    }
  }
>
  <svg
    className="recharts-surface"
    height={300}
    version="1.1"
    viewBox="0 0 500 300"
    width={500}
  >
    <defs>
      <clipPath
        id="recharts2-clip"
      >
        <rect
          height={260}
          width={390}
          x={80}
          y={5}
        />
      </clipPath>
    </defs>
    <g
      className="recharts-cartesian-grid"
    >
      <g
        className="recharts-cartesian-grid-horizontal"
      >
        <line
          fill="none"
          height={260}
          stroke="#ccc"
          strokeDasharray="3 3"
          width={390}
          x={80}
          x1={80}
          x2={470}
          y={5}
          y1={265}
          y2={265}
        />
        <line
          fill="none"
          height={260}
          stroke="#ccc"
          strokeDasharray="3 3"
          width={390}
          x={80}
          x1={80}
          x2={470}
          y={5}
          y1={200}
          y2={200}
        />
        <line
          fill="none"
          height={260}
          stroke="#ccc"
          strokeDasharray="3 3"
          width={390}
          x={80}
          x1={80}
          x2={470}
          y={5}
          y1={135}
          y2={135}
        />
        <line
          fill="none"
          height={260}
          stroke="#ccc"
          strokeDasharray="3 3"
          width={390}
          x={80}
          x1={80}
          x2={470}
          y={5}
          y1={70}
          y2={70}
        />
        <line
          fill="none"
          height={260}
          stroke="#ccc"
          strokeDasharray="3 3"
          width={390}
          x={80}
          x1={80}
          x2={470}
          y={5}
          y1={5}
          y2={5}
        />
      </g>
      <g
        className="recharts-cartesian-grid-vertical"
      >
        <line
          fill="none"
          height={260}
          stroke="#ccc"
          strokeDasharray="3 3"
          width={390}
          x={80}
          x1={177.5}
          x2={177.5}
          y={5}
          y1={5}
          y2={265}
        />
        <line
          fill="none"
          height={260}
          stroke="#ccc"
          strokeDasharray="3 3"
          width={390}
          x={80}
          x1={372.5}
          x2={372.5}
          y={5}
          y1={5}
          y2={265}
        />
        <line
          fill="none"
          height={260}
          stroke="#ccc"
          strokeDasharray="3 3"
          width={390}
          x={80}
          x1={80}
          x2={80}
          y={5}
          y1={5}
          y2={265}
        />
        <line
          fill="none"
          height={260}
          stroke="#ccc"
          strokeDasharray="3 3"
          width={390}
          x={80}
          x1={470}
          x2={470}
          y={5}
          y1={5}
          y2={265}
        />
      </g>
    </g>
    <g
      className="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
    >
      <line
        className="recharts-cartesian-axis-line"
        fill="none"
        height={30}
        stroke="#666"
        width={390}
        x={80}
        x1={80}
        x2={470}
        y={265}
        y1={265}
        y2={265}
      />
      <g
        className="recharts-cartesian-axis-ticks"
      >
        <g
          className="recharts-layer recharts-cartesian-axis-tick"
        >
          <line
            className="recharts-cartesian-axis-tick-line"
            fill="none"
            height={30}
            stroke="#666"
            width={390}
            x={80}
            x1={177.5}
            x2={177.5}
            y={265}
            y1={271}
            y2={265}
          />
          <text
            className="recharts-text recharts-cartesian-axis-tick-value"
            fill="#666"
            height={30}
            stroke="none"
            textAnchor="middle"
            width={390}
            x={177.5}
            y={273}
          >
            <tspan
              dy="0.71em"
              x={177.5}
            >
              solar
            </tspan>
          </text>
        </g>
        <g
          className="recharts-layer recharts-cartesian-axis-tick"
        >
          <line
            className="recharts-cartesian-axis-tick-line"
            fill="none"
            height={30}
            stroke="#666"
            width={390}
            x={80}
            x1={372.5}
            x2={372.5}
            y={265}
            y1={271}
            y2={265}
          />
          <text
            className="recharts-text recharts-cartesian-axis-tick-value"
            fill="#666"
            height={30}
            stroke="none"
            textAnchor="middle"
            width={390}
            x={372.5}
            y={273}
          >
            <tspan
              dy="0.71em"
              x={372.5}
            >
              wind
            </tspan>
          </text>
        </g>
      </g>
    </g>
    <g
      className="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
    >
      <line
        className="recharts-cartesian-axis-line"
        fill="none"
        height={260}
        stroke="#666"
        width={60}
        x={20}
        x1={80}
        x2={80}
        y={5}
        y1={5}
        y2={265}
      />
      <g
        className="recharts-cartesian-axis-ticks"
      >
        <g
          className="recharts-layer recharts-cartesian-axis-tick"
        >
          <line
            className="recharts-cartesian-axis-tick-line"
            fill="none"
            height={260}
            stroke="#666"
            width={60}
            x={20}
            x1={74}
            x2={80}
            y={5}
            y1={265}
            y2={265}
          />
          <text
            className="recharts-text recharts-cartesian-axis-tick-value"
            fill="#666"
            height={260}
            stroke="none"
            textAnchor="end"
            width={60}
            x={72}
            y={265}
          >
            <tspan
              dy="0.355em"
              x={72}
            >
              0
            </tspan>
          </text>
        </g>
        <g
          className="recharts-layer recharts-cartesian-axis-tick"
        >
          <line
            className="recharts-cartesian-axis-tick-line"
            fill="none"
            height={260}
            stroke="#666"
            width={60}
            x={20}
            x1={74}
            x2={80}
            y={5}
            y1={200}
            y2={200}
          />
          <text
            className="recharts-text recharts-cartesian-axis-tick-value"
            fill="#666"
            height={260}
            stroke="none"
            textAnchor="end"
            width={60}
            x={72}
            y={200}
          >
            <tspan
              dy="0.355em"
              x={72}
            >
              20
            </tspan>
          </text>
        </g>
        <g
          className="recharts-layer recharts-cartesian-axis-tick"
        >
          <line
            className="recharts-cartesian-axis-tick-line"
            fill="none"
            height={260}
            stroke="#666"
            width={60}
            x={20}
            x1={74}
            x2={80}
            y={5}
            y1={135}
            y2={135}
          />
          <text
            className="recharts-text recharts-cartesian-axis-tick-value"
            fill="#666"
            height={260}
            stroke="none"
            textAnchor="end"
            width={60}
            x={72}
            y={135}
          >
            <tspan
              dy="0.355em"
              x={72}
            >
              40
            </tspan>
          </text>
        </g>
        <g
          className="recharts-layer recharts-cartesian-axis-tick"
        >
          <line
            className="recharts-cartesian-axis-tick-line"
            fill="none"
            height={260}
            stroke="#666"
            width={60}
            x={20}
            x1={74}
            x2={80}
            y={5}
            y1={70}
            y2={70}
          />
          <text
            className="recharts-text recharts-cartesian-axis-tick-value"
            fill="#666"
            height={260}
            stroke="none"
            textAnchor="end"
            width={60}
            x={72}
            y={70}
          >
            <tspan
              dy="0.355em"
              x={72}
            >
              60
            </tspan>
          </text>
        </g>
        <g
          className="recharts-layer recharts-cartesian-axis-tick"
        >
          <line
            className="recharts-cartesian-axis-tick-line"
            fill="none"
            height={260}
            stroke="#666"
            width={60}
            x={20}
            x1={74}
            x2={80}
            y={5}
            y1={5}
            y2={5}
          />
          <text
            className="recharts-text recharts-cartesian-axis-tick-value"
            fill="#666"
            height={260}
            stroke="none"
            textAnchor="end"
            width={60}
            x={72}
            y={5}
          >
            <tspan
              dy="0.355em"
              x={72}
            >
              80
            </tspan>
          </text>
        </g>
      </g>
    </g>
    <g
      className="recharts-layer recharts-bar"
    >
      <g
        className="recharts-layer recharts-bar-rectangles"
        clipPath={null}
      >
        <g
          className="recharts-layer"
        >
          <g
            className="recharts-layer recharts-bar-rectangle"
          />
          <g
            className="recharts-layer recharts-bar-rectangle"
          />
        </g>
      </g>
    </g>
  </svg>
  <div
    className="recharts-legend-wrapper"
    style={
      Object {
        "bottom": 5,
        "height": "auto",
        "left": 20,
        "position": "absolute",
        "width": 450,
      }
    }
  >
    <ul
      className="recharts-default-legend"
      style={
        Object {
          "margin": 0,
          "padding": 0,
          "textAlign": "center",
        }
      }
    >
      <li
        className="recharts-legend-item legend-item-0"
        style={
          Object {
            "display": "inline-block",
            "marginRight": 10,
          }
        }
      >
        <svg
          className="recharts-surface"
          height={14}
          style={
            Object {
              "display": "inline-block",
              "marginRight": 4,
              "verticalAlign": "middle",
            }
          }
          version="1.1"
          viewBox="0 0 32 32"
          width={14}
        >
          <path
            className="recharts-legend-icon"
            d="M0,4h32v24h-32z"
            fill="#82ca9d"
            stroke="none"
          />
        </svg>
        <span
          className="recharts-legend-item-text"
        >
          perc
        </span>
      </li>
    </ul>
  </div>
  <div
    className="recharts-tooltip-wrapper"
    style={
      Object {
        "MozTransform": "translate(undefinedpx, undefinedpx)",
        "OTransform": "translate(undefinedpx, undefinedpx)",
        "WebkitTransform": "translate(undefinedpx, undefinedpx)",
        "msTransform": "translate(undefinedpx, undefinedpx)",
        "pointerEvents": "none",
        "position": "absolute",
        "top": 0,
        "transform": "translate(undefinedpx, undefinedpx)",
        "visibility": "hidden",
      }
    }
  >
    <div
      className="recharts-default-tooltip"
      style={
        Object {
          "backgroundColor": "#fff",
          "border": "1px solid #ccc",
          "margin": 0,
          "padding": 10,
          "whiteSpace": "nowrap",
        }
      }
    >
      <p
        className="recharts-tooltip-label"
        style={
          Object {
            "margin": 0,
          }
        }
      >
        
      </p>
    </div>
  </div>
</div>
`;
